### PR TITLE
feat(ml): Visformer architecture with torch backend

### PR DESCRIPTION
**Pull request by beniz**
This an implementation of https://arxiv.org/pdf/2104.12533.pdf with Visformer-tiny and Visformer-S, available respectively via API:

```
"mllib":{
  "template":"visformer",
  "template_params": 
   {
     "visformer_flavor":"visformer_tiny",
     "dropout": 0.1
   }
}
```

Quick training results on cats/dogs yield much easier and natural convergence than with ViT, probably thanks to the removal of the lowest attention layer, staging and batch norm, as elegantly detailed in the paper.

image

### CODE REVIEW

**Review by fantes — APPROVED** [review]

**Review by Bycob — COMMENTED** [review]
Add template in documentation?

**Comment by beniz** [comment]
> Add template in documentation?

done.

File: tests/ut-torchapi.cc [hunk 1]
```diff
@@ -2908,4 +2908,93 @@ TEST(torchapi, service_train_vit_images_multigpu)
...
+  std::string jstr
+      = "{\"mllib\":\"torch\",\"description\":\"image\",\"type\":"
+        "\"supervised\",\"model\":{\"repository\":\""
+
+        + vit_train_repo
+        + "\"},\"parameters\":{\"input\":{\"connector\":\"image\","
+          "\"width\":224,\"height\":224,\"db\":true},\"mllib\":{\"nclasses\":"
+          "2,\"template\":\"visformer\",\"gpu\":true,\"template_params\":{"
+          "\"vit_"
+          "flavor\":\"visformer_tiny\"}}}}";
```
Comment (Bycob) [inline tests/ut-torchapi.cc:2932]: "vit_flavor" or "visformer_flavor"?

Comment (beniz) [reply tests/ut-torchapi.cc:2932]: good catch, thanks!

Comment (beniz) [reply tests/ut-torchapi.cc:2932]: fixed

**Review by Bycob — APPROVED** [review]
